src/firestore/firekit.ts

```diff
@@ -773,6 +792,97 @@ export class RoarFirekit {
     }
   }
 
+  async createUser(
+    email: string,
+    password: string,
+    userData: ICreateUserInput,
+  ) {
+    this._verify_authentication();
+
+    const isEmailAvailable = await this.isUsernameAvailable(email)
+    if(isEmailAvailable){
+      let userObject: any = {
+        userType: "student",
+        studentData: {}
+      }
+      if(!_get(userData, 'age') && !_get(userData, 'dob')) {
+        // Throw exception or return code
+        // User can NOT lack date of birth AND age
+        return 1;
+      }
+      if(_get(userData, 'name')) userObject['name'] = userData.name;
+      if(_get(userData, 'dob')) userObject['studentData']['dob'] = userData.dob;
+      if(_get(userData, 'age')){
+        const age: number = Number(userData.age);
+        const yearOffset = Math.floor(age);
+        const monthOffset = age % yearOffset;
+        let calcDob = new Date();
+        calcDob.setFullYear(calcDob.getFullYear() - yearOffset);
+        calcDob.setMonth(calcDob.getMonth() - monthOffset)
+        userObject['studentData']['dob'] = calcDob;
+      }
+      if(_get(userData, 'gender')) userObject['studentData']['gender'] = userData.gender;
+      if(_get(userData, 'ell_status')) userObject['studentData']['ell_status'] = userData.ell_status;
+      if(_get(userData, 'iep_status')) userObject['studentData']['iep_status'] = userData.iep_status;
+      if(_get(userData, 'frl_status')) userObject['studentData']['frl_status'] = userData.frl_status;
+
+      const dateNow = Date.now()
+      // create district entry
+      const districtId = _get(userData, 'district');
+      if(districtId) {
+        userObject['districts'] = {
+          current: [districtId],
+          all: [districtId],
+          dates: {
+            [districtId!]: {
+              from: dateNow,
+              to: null
+            }
+          }
+        }
+      }
+      // create school entry
+      const schoolId = _get(userData, 'school');
+      if(schoolId){
+        userObject['schools'] = {
+          current: [schoolId],
+          all: [schoolId],
+          dates: {
+            [schoolId!]: {
+              from: dateNow,
+              to: null
+            }
+          }
+        }
+      }
+      // create class entry
+      const classId = _get(userData, 'class');
+      if(classId){
+        userObject['classes'] = {
+          current: [classId],
+          all: [classId],
+          dates: {
+            [classId!]: {
+              from: dateNow,
+              to: null
+            }
+          }
+        }
+      }
+      const cloudCreateUser = httpsCallable(this.admin.functions, 'createUser');
+      const adminId = await cloudCreateUser({email, password, userData});
```

Comment:
`cloudCreateUser` will return an object like

```
adminId = {
    data: {
        adminUid: "adminUid",
    }
};
```

So you can use something like
```
const adminResponse = await cloudCreateUser({email, password, userData});
const adminUid = _get(adminResponse.data, "adminUid");

const cloudCreateAppUser = httpsCallable(this.app.functions, 'createUser');
const appResponse = await cloudCreateAppUser({email, password, userData, adminUid});
const assessmentUid = _get(appResponse.data, "assessmentUid");

// Inside of the assessment cloud function, be sure to set UID claims since you will now have both UIDs.
// Then call yet another admin firebase cloud function to add the assessment UID to the user doc in Firestore and also call the set UID claims function.